src/connection.cc

```diff
@@ -46,37 +46,52 @@ bool Connection::handle_read(const boost::system::error_code& error,
     return false; // error
   }
 
+  // TODO: tentative if this is good practice for sending the string
+  // std::string buffer_string = std::string(buffer_);
+  std::string buffer_string;
+  std::copy(buffer_.begin(), buffer_.begin()+bytes_transferred, std::back_inserter(buffer_string));
+
   // RequestParser will insert data into the request struct
-  Request request;
-  // TODO: use the parser to parse the request.
-  // std::tie(result, std::ignore) = request_parser_.parse(
-  //     request, buffer_.data(), buffer_.data() + bytes_transferred);
-  // The line under this is meant to keep it compiling.
-  // result = Request::good;
-
-  // if (result == Request::good) {
-    // You'll need this: RequestHandler request_handler;
-    // Need to iterate through handler_id possibilities by longest prefix
-    // So... maybe add a for loop or something?
-    // std::string handler_id = request.uri(); // recommend changing name to handler_id in Request struct
-    // for(auto const &e : *path_to_info_) {
-      // if(e.first == handler_id) {
-        // Instantiate a handler --> see https://piazza.com/class/iwy6115h8ce37q?cid=68
-        // request_handler = RequestHandler::???(handler_id);
-        // break;
-      // }
-    // }
-    // detect if request_handler never got initialized
-    // if so set it to a default handler, or maybe just return false ?
-    Response response;
-    // request_handler->handle_request(request, response);
+  std::unique_ptr<Request> request = Request::Parse(buffer_string);
+
+  // TODO: Decide if we want to check "status" to see if it parsed correctly...
+  // If so, we may need to make the enum result_type public or play with 0, 1, 2 values
+
+  // TODO: problem - can't use an abstract handler... maybe default to echo handler? 
+  // RequestHandler request_handler;
+
+  std::string handler_id = request->uri();
+    
+  // Need to iterate through handler_id possibilities by longest prefix
+  // TODO: Decide if we want to limit handler_id to "/" or ""
+  while (handler_id != "") {
+    // Check the map to see if it holds handler_id
+    std::map<std::string, PathInfo>::iterator found_info_iterator = path_to_info_->find(handler_id);
```

Comment:
I think it's simpler to use `string::count` than an iterator and `string::find`
http://www.cplusplus.com/reference/unordered_map/unordered_map/count/